client/src/areas/grail/GrailArea.tsx

```diff
@@ -173,20 +185,36 @@ const GrailAreaInternal: FC<Props> = props => {
 
   function setThemeAndTitle() {
     let theme: IAppTheme = null;
+
+    let defaultThemeToSet = defaultTheme;
+    let ethThemeToSet = ethTheme;
+    let runewordThemeToSet = runewordTheme;
+
+    if (darkThemeIsEnabled()) {
+      defaultThemeToSet = defaultThemeDark;
+      ethThemeToSet = ethThemeDark;
+      runewordThemeToSet = runewordThemeDark;
+    }
+
     switch (grailMode) {
       case GrailMode.Eth:
-        theme = ethTheme;
+        theme = ethThemeToSet;
```

Comment:
to address your DRY comment.
Only define all the themes once (if all colors are really identical). => leave it as it was

what you could do is simply this:

```

function getTheme(theme: IAppTheme , useDarkMode?: boolean) {
  return { ...theme, ...(useDarkMode && {type: 'dark'})};
}

```

this basically copies the theme config and then if requireds adds the dark type.

and then in the switch you just call it like this:

```
case GrailMode.Eth:
        theme = getTheme(ethTheme, darkThemeIsEnabled()) ;	   
```

Reply:
even better, do not implement the `getTheme` method here, but in `App.tsx`.
So the code in here you can leave almost the same except that you pass to the `setAppTheme` method if it should use the dark theme:

`setAppTheme(theme, useDarkMode);` 

and then you do what I wrote about within this method :) (you have to update the signature of the method in the corresponding interface)